src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/jobsboard/api/ApplicationTestCase.kt

```diff
@@ -116,7 +121,8 @@ abstract class ApplicationTestCase {
   @BeforeEach
   fun setup() {
     whenever(timeProvider.now()).thenCallRealMethod()
-    whenever(dateTimeProvider.now).thenReturn(Optional.of(jobCreationTime))
+    whenever(dateTimeProvider.now).thenReturn(Optional.of(defaultCurrentTime))
+    countOfGettingCurrentTime[0] = 0
```

Comment:
I like your refactor, @rickchoijd; I have just one question regarding this new counter you have added. It's like a gut feeling that we're maintaining an external attribute to control the state of the mock. Do you know if the mock itself can provide that counter so we don't need an 'external' variable to maintain that control?

Reply:
That API call cannot be used within the mocked function (lambda in thenAnswer{} )
It gave me _**stack-overflow**_ (circular reference)
ref:  https://github.com/mockito/mockito/issues/77

On the other hand, the InvocationOnMock (from thenAnswer()) lacked call-count. (There is sequence number in internal class, but it is not invocation counter)

Alternatively, we can write our own invocation counter helper class, to formalise the usage (still, storing outside mocking)
something similar to this: https://stackoverflow.com/questions/25543250/getting-number-of-calls-to-a-mock
any preference?

Reply:
I prefer to avoid helper classes as much as possible; I think the danger is to end up with anaemic domain.

A solution could be to wrap the mock so that our object is encapsulated with all the functionality and attributes we need.

But because this PR is taking a bit of time, I propose just leaving it out for now and writing a specific task. We can refine it, think about options, and then implement it.

What do you think?

Reply:
raised this "Refactor Timestamp mocking" on MIro board, thanks.